ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/util/SlotProcessorUtil.java

```diff
@@ -15,45 +15,28 @@
 
 import static java.lang.Math.toIntExact;
 import static tech.pegasys.artemis.datastructures.Constants.LATEST_BLOCK_ROOTS_LENGTH;
-import static tech.pegasys.artemis.datastructures.Constants.LATEST_RANDAO_MIXES_LENGTH;
 
 import com.google.common.primitives.UnsignedLong;
 import java.util.List;
 import net.consensys.cava.bytes.Bytes32;
 import tech.pegasys.artemis.datastructures.Constants;
-import tech.pegasys.artemis.datastructures.blocks.BeaconBlock;
 import tech.pegasys.artemis.datastructures.state.BeaconState;
 import tech.pegasys.artemis.datastructures.util.BeaconStateUtil;
 import tech.pegasys.artemis.statetransition.StateTransitionException;
 
 public class SlotProcessorUtil {
-  public static void updateLatestRandaoMixes(BeaconState state) {
-    // TODO: change values to UnsignedLong
-    int currSlot = state.getSlot().intValue();
-    List<Bytes32> latestRandaoMixes = state.getLatest_randao_mixes();
-    int index = (currSlot - 1) % LATEST_RANDAO_MIXES_LENGTH;
-    Bytes32 prevSlotRandaoMix = latestRandaoMixes.get(index);
-    latestRandaoMixes.set(currSlot % LATEST_RANDAO_MIXES_LENGTH, prevSlotRandaoMix);
-  }
 
-  public static void updateRecentBlockHashes(BeaconState state, BeaconBlock block)
+  public static void updateBlockRoots(BeaconState state, Bytes32 previous_block_root, String logPrefix)
```

Comment:
I believe that this method is superior to what is currently in master bc you are referencing the real previous_block.